app/controllers/api/results_controller.rb

```diff
@@ -6,6 +6,77 @@ def results_search_conditions
       MiqReportResult.for_user(User.current_user).where_clause.ast
     end
 
+    def apply_limit_and_offset(results, options)
+      results.slice(options['offset'].to_i, options['limit'].to_i) || []
+    end
+
+    def sort_params_from(report_result)
+      params['sort_order'] ||= :ascending
+      params['sort_order'] = :descending if params['sort_order'] == 'desc'
+
+      sort_colums = sort_by(report_result).split(",").collect do |attr|
+        if report(report_result).col_order&.include?(attr)
+          attr
+        else
+          raise BadRequestError, "#{attr} is not a valid attribute for #{report_result.name}"
+        end
+      end.compact
+
+      {:sort_by => sort_colums, :sort_order => (params['sort_order'] || :ascending).to_sym }
+    end
+
+    def param_result_set?
+      params.key?(:hash_attribute) && params[:hash_attribute] == "result_set"
+    end
+
+    def format_result_set(miq_report, result_set)
+      tz = miq_report.get_time_zone(Time.zone)
+
+      col_format_hash = miq_report.col_order.zip(miq_report.col_formats).to_h
+
+      result_set.map! do |row|
+        row.map do |key, _|
+          [key, miq_report.format_column(key, row, tz, col_format_hash[key])]
+        end.to_h
+      end
+    end
+
+    def report(report_result)
+      @report ||= report_result.report || report_result.miq_report
+    end
+
+    def sort_by(report_result)
```

Comment:
is there a reason to join here?
seems we just `split` it once it comes out

Reply:
no, we need to pass array, thanks!